### PR TITLE
Adds asserts messages

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/algorithm/searching.d [hunk 1]
```diff
@@ -848,7 +848,8 @@ if (isForwardRange!R
     }
 
     //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
-    static if (isInfinite!R) assert(0);
+    static if (isInfinite!R) assert(false, R.stringof ~ "must be an infinite"
```
Comment (thewilsonator) [inline std/algorithm/searching.d]: must _not_ be.

Comment (thewilsonator) [reply std/algorithm/searching.d]: also just `static assert(!isInfinite!R, msg)`

File: std/algorithm/searching.d [hunk 2]
```diff
@@ -951,7 +952,8 @@ if (isInputRange!R &&
     }
 
     //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
-    static if (isInfinite!R) assert(0);
+    static if (isInfinite!R) assert(false, R.stringof ~ " must be an inifite"
```
Comment (thewilsonator) [inline std/algorithm/searching.d]: ditto.

File: std/algorithm/searching.d (hunk 1 above)
Comment (burner) [reply std/algorithm/searching.d]: that will no compile, I tried. depending on the type of R that function will not have a return and an assert(false) at the end.

Comment (thewilsonator) [reply std/algorithm/searching.d]: Oh, I misread that as a static assert. My bad.

**Comment by burner** [comment]
@thewilsonator I was a second to slow

File: std/algorithm/searching.d [hunk 3]
```diff
@@ -848,7 +848,8 @@ if (isForwardRange!R
     }
 
     //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
-    static if (isInfinite!R) assert(0);
+    static if (isInfinite!R) assert(false, R.stringof ~ "must not be an"
```
Comment (shove70) [inline std/algorithm/searching.d]: static if (isInfinite!R) assert(false, R.stringof ~ " must not be an"
Lost a space. ^_^

**Comment by burner** [comment]
@thewilsonator could you have a look?

**Comment by thewilsonator** [comment]
Done.

**Comment by burner** [comment]
that was quick, how do you do it? thanks

**Comment by thewilsonator** [comment]
[magic](https://media2.giphy.com/media/12NUbkX6p4xOO4/giphy.gif?cid=790b76115d133d727034654d36fb1161&rid=giphy.gif)

**Comment by burner** [comment]
yeah, that is what I expected :-)

**Comment by thewilsonator** [comment]
You know me too well.

**Comment by burner** [comment]
thanks

**Pull request merged**